### PR TITLE
add SetDefaults from parent (using context), change Destination to us…

### PR DESCRIPTION
**Pull request by vaikas**
…e KnativeReference

### CODE REVIEW

File: apis/duck/v1/destination_test.go [hunk 1]
```diff
@@ -183,3 +197,57 @@ func TestDestination_GetRef(t *testing.T) {
 		})
 	}
 }
+
+func TestDestinationSetDefaults(t *testing.T) {
+	ctx := context.TODO()
```
Comment (vagababov) [inline apis/duck/v1/destination_test.go]:
```suggestion
	ctx := context.Background()
```

File: apis/duck/v1/destination_test.go [hunk 2]
```diff
@@ -183,3 +197,57 @@ func TestDestination_GetRef(t *testing.T) {
...
+			d:    &Destination{Ref: &KnativeReference{}, URI: apis.HTTP("example.com")},
+			ctx:  apis.WithinParent(ctx, metav1.ObjectMeta{Namespace: parentNamespace}),
+			want: parentNamespace,
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc.d.SetDefaults(tc.ctx)
+			if tc.d.Ref != nil && tc.d.Ref.Namespace != tc.want {
+				t.Errorf("%s: got: %s wanted %s", name, tc.d.Ref.Namespace, tc.want)
```
Comment (vagababov) [inline apis/duck/v1/destination_test.go]:
```suggestion
				t.Errorf("Got: %s wanted %s", tc.d.Ref.Namespace, tc.want)
```

File: apis/duck/v1/destination_test.go [hunk 3]
```diff
@@ -183,3 +197,57 @@ func TestDestination_GetRef(t *testing.T) {
...
+			}
+			if tc.d.Ref == nil && tc.want != "" {
+				t.Errorf("%s: got: nil Ref wanted %s", name, tc.want)
```
Comment (vagababov) [inline apis/duck/v1/destination_test.go]:
```suggestion
				t.Errorf("Got: nil Ref wanted %s",  tc.want)
```

File: apis/duck/v1/destination_test.go (hunk 2 above)
Comment (n3wscott) [reply apis/duck/v1/destination_test.go]: it is helpful to have the test name in the test output

File: apis/duck/v1/destination_test.go (hunk 3 above)
Comment (n3wscott) [reply apis/duck/v1/destination_test.go]: booo

File: apis/duck/v1/destination_test.go (hunk 2 above)
Comment (vagababov) [reply apis/duck/v1/destination_test.go]: When you use `t.Run` the test name is right there, above :-)

**Pull request merged**